---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# omicsTools

<!-- badges: start -->
[![CRAN status](https://www.r-pkg.org/badges/version/omicsTools)](https://cran.r-project.org/package=omicsTools)
[![Lifecycle: experimental](https://img.shields.io/badge/lifecycle-experimental-orange.svg)](https://lifecycle.r-lib.org/articles/stages.html#experimental)

<!-- badges: end -->

The goal of omicsTools is to provide tools for processing and analyzing omics data from genomics, transcriptomics, proteomics, and metabolomics platforms. It provides functions for preprocessing, normalization, visualization, and statistical analysis, as well as machine learning algorithms for predictive modeling. omicsTools is an essential tool for researchers working with high-throughput omics data in fields such as biology, bioinformatics, and medicine.

License: AGPL-3.0


## Install dependencies

```{r}
if (!require("BiocManager", quietly = TRUE)) { install.packages("BiocManager") }
if (!require("pvca", quietly = TRUE)) { BiocManager::install("pvca") }
```

## Installation

### CRAN version

You can install the Stable version of omicsTools like so:

```{r}
install.packages("omicsTools")
```

### Development version

To get a bug fix, or use a feature from the development version, you can install omicsTools from GitHub.
```{r}
if (!require("devtools", quietly = TRUE))
    install.packages("devtools")
devtools::install_github("cheemalab/omicsTools")
```



## Example of imputation

```{r}
# Load the CSV data
data_file <- system.file("extdata", "example1.csv", package = "omicsTools")
data <- readr::read_csv(data_file)
# Apply the impute function
imputed_data <- omicsTools::handle_missing_values(data)
# Write the imputed data to a new CSV file
readr::write_csv(imputed_data, paste0(tempdir(), "/imputed_data.csv"))
```


## Example of QC-normalization


```{r}
# Load the CSV data
data_file <- system.file("extdata", "example2.csv", package = "omicsTools")
data <- readr::read_csv(data_file)
# Apply the normalize function
normalized_data <- omicsTools::qc_normalize(data)
# Write the normalized data to a new CSV file
readr::write_csv(normalized_data, paste0(tempdir(), "/normalized_data.csv"))
```


## Anomaly removal Example 1: Using generated data

```{r}
# Using generated data
library(omicsTools)
omics_data <- createOmicsData()

# Define custom thresholds
custom_thresholds <- define_thresholds(skewness = 3, kurtosis = 8)

# Update the OmicsData object with custom thresholds
omics_data@thresholds <- custom_thresholds

# Calculate measures for each feature
omics_data <- calculate_measures(omics_data)

# Flag anomalies
omics_data <- flag_anomalies(omics_data)

# Plotting distribution measures
plot_distribution_measures(omics_data)

# Plotting sample measures
plot_sample_measures(omics_data)

# Prepare data for UpSet plot
upset_data <- prepare_upset_data(omics_data)

# Convert to binary matrix for UpSetR
upset_matrix <- convert_to_binary_matrix(upset_data)

# Ensure there are at least two columns for the UpSet plot
ensure_enough_sets_for_upset(upset_matrix)
```

# Anomaly removal Example 2: Using user-provided data

```{r}

library(omicsTools)

# Assuming user_data is a data frame where rows are samples and columns are features
user_data <- generate_data_with_anomalies(n_samples = 100, n_features = 200)  # Replace with actual data loading if available
omics_data_user <- createOmicsData(data = user_data)

# Define custom thresholds
custom_thresholds_user <- define_thresholds(skewness = 4, kurtosis = 6, shapiro_p = 1e-5, cooks_distance = 0.5, lof = 10)

# Update the OmicsData object with custom thresholds
omics_data_user@thresholds <- custom_thresholds_user

# Calculate measures for each feature
omics_data_user <- calculate_measures(omics_data_user)

# Flag anomalies
omics_data_user <- flag_anomalies(omics_data_user)

# Plotting distribution measures
plot_distribution_measures(omics_data_user)

# Plotting sample measures
plot_sample_measures(omics_data_user)

# Prepare data for UpSet plot
upset_data_user <- prepare_upset_data(omics_data_user)

# Convert to binary matrix for UpSetR
upset_matrix_user <- convert_to_binary_matrix(upset_data_user)

# Ensure there are at least two columns for the UpSet plot
ensure_enough_sets_for_upset(upset_matrix_user)
```

### Code style

Since this is a collaborative project, please adhere to the following code formatting conventions:
* We use the tidyverse style guide (https://style.tidyverse.org/)
* Please write roxygen2 comments as full sentences, starting with a capital letter and ending with a period. Brevity is preferred (e.g., "Calculates standard deviation" is preferred over "This method calculates and returns a standard deviation of given set of numbers").
